### ISSUE TITLE
Argument of type is not assignable to parameter of type 'ISendMailOptions'

### ISSUE THREAD

**Issue by MegaphoneCA** [nest-modules/mailer#25]
Hello, trying to update the package to use with NestJS V6

Version: "@nest-modules/mailer": "^1.1.2"

Here is my config
```
        imports: [
            MailerModule.forRoot(
                {
                    transport: `smtps://${process.env.MAIL_USER}:${process.env.MAIL_PASS}@${process.env.MAIL_HOST}`,
                    defaults:
                    {
                        from: `"System Administrator" <${process.env.MAIL_FROM}>`,
                    }
                }
            )
        ],
```

Here is where the error is occuring
```
        return this.mailerService.sendMail(
            {
                to: user.email,
                subject,
                text,
            }
        );
```

Here is the error I am receiving from the TS compiler:
```
src/user/user.service.ts:51:17 - error TS2345: Argument of type '{ to: string; subject: string; text: string; }' is not assignable to parameter of type 'ISendMailOptions'.
  Object literal may only specify known properties, and 'to' does not exist in type 'ISendMailOptions'.

51                 to: user.email,
                   ~~~~~~~~~~~~~~

    at createTSError (C:\Users\USERNAME\Development\xhauler\server\node_modules\ts-node\src\index.ts:226:12)
    at getOutput (C:\Users\USERNAME\Development\xhauler\server\node_modules\ts-node\src\index.ts:335:40)
    at Object.compile (C:\Users\USERNAME\Development\xhauler\server\node_modules\ts-node\src\index.ts:368:11)
    at Module.m._compile (C:\Users\USERNAME\Development\xhauler\server\node_modules\ts-node\src\index.ts:414:43)
    at Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Object.require.extensions.(anonymous function) [as .ts] (C:\Users\USERNAME\Development\xhauler\server\node_modules\ts-node\src\index.ts:417:12)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Module.require (internal/modules/cjs/loader.js:636:17)
```

Example was working earlier with Nest 5.4 and V1.0.3

Am I missing a step in the migration process?

Regards

**Comment by mfeldmeier** [comment]
Hello, 

I have the same Problem. 
In the new version is the ISendMailOptions required but there is no params for: To, Cc, Bcc,  From. 
So be careful to Update to the Nestjs version 6.x because the Mailer isn't working at the moment.

Any solutions?

**Comment by mfeldmeier** [comment]
@MegaphoneCA super strange I just removed the class for  sendMailOptions:
``` TS
// from
import { SentMessageInfo } from 'nodemailer';
/** Interfaces **/
import { MailerOptions } from './interfaces/mailer-options.interface';
import { ISendMailOptions } from './interfaces/send-mail-options.interface';
export declare class MailerService {
    private readonly mailerOptions;
    private transporter;
    constructor(mailerOptions: MailerOptions);
    sendMail(sendMailOptions: ISendMailOptions): Promise<SentMessageInfo>;
}
```

``` TS
// to
import { SentMessageInfo } from 'nodemailer';
/** Interfaces **/
import { MailerOptions } from './interfaces/mailer-options.interface';
import { ISendMailOptions } from './interfaces/send-mail-options.interface';
export declare class MailerService {
    private readonly mailerOptions;
    private transporter;
    constructor(mailerOptions: MailerOptions);
    sendMail(sendMailOptions): Promise<SentMessageInfo>;
}
```

After reloading Nest App it should be working than. 
After this you can add the Class to the Options and reload again and it still working. 

Maybe it's a caching problem? 
I uninstalled the Package and install it again and its still working. 

How?

**Comment by MegaphoneCA** [comment]
I'm confused, with your instructions... Can you clarify exactly what steps I need to take?

**Comment by mfeldmeier** [comment]
Go to the mailer.service.d.ts : 
```TS
import { SentMessageInfo } from 'nodemailer';
/** Interfaces **/
import { MailerOptions } from './interfaces/mailer-options.interface';
import { ISendMailOptions } from './interfaces/send-mail-options.interface';
export declare class MailerService {
    private readonly mailerOptions;
    private transporter;
    constructor(mailerOptions: MailerOptions);
    sendMail(sendMailOptions: ISendMailOptions): Promise<SentMessageInfo>;
}

```
In the sendMail function is the parameter sendMailOptions. This is from Type `ISendMailOptions`
Remove this at first, save the file and reload your application. 

Now you shouldn't get any error. 

Go again to this file and add `ISendMailOptions` to the sendMailOptions.
```TS
sendMailOptions:ISendMailOptions
```
 Save again and reload your application. 
This works for me and all is working now. 

Is that easier to understand?

**Comment by MegaphoneCA** [comment]
That wasn't the part I found confusing. I am hoping for a fix that will be ingegrated into the project and not something I have to do myself.

Will the code you shared be integrated into the NPM version? If I change the code in my node_modules folder than it is only local and won't carry over to my production environment.

**Comment by mfeldmeier** [comment]
It's weird. 
Because I change it and put it back again. 
It looks like a bug but it would work. 
So I don't know if or how you can fix something like that in an npm module. 

Maybe this is just a temporary fix.

**Comment by ahmadarif** [comment]
I've the same problem, and downgrade to **v1.0.4** :D

**Comment by justinlazaro-ubidy** [comment]
i have a same issue here

**Comment by justinlazaro-ubidy** [comment]
temporary fix.

![image](https://user-images.githubusercontent.com/33537770/55673600-96596080-58dc-11e9-8a7d-45d3b1baef2e.png)

**Comment by nelsonBlack** [comment]
> temporary fix.
> 
> ![image](https://user-images.githubusercontent.com/33537770/55673600-96596080-58dc-11e9-8a7d-45d3b1baef2e.png)

working temporary fix for sendmail options in NestJS V6, hope fix is added to support v6

**Comment by cdiaz** [comment]
Thanks, sorry for the delay, it is already fixed in 1.1.3 version

**Comment by justinlazaro-ubidy** [comment]
@cdiaz thanks
